packages/router/router.ts

```diff
@@ -828,11 +836,32 @@ export function createRouter(init: RouterInit): Router {
       }
     );
 
-    // Kick off initial data load if needed.  Use Pop to avoid modifying history
-    if (!state.initialized) {
+    if (state.initialized) {
+      return router;
+    }
+
+    let lazyMatches = state.matches.filter((m) => m.route.lazy);
+
+    if (lazyMatches.length === 0) {
+      // Kick off initial data load if needed.  Use Pop to avoid modifying history
       startNavigation(HistoryAction.Pop, state.location);
+      return router;
     }
 
+    // Load lazy modules, then kick off initial data load if needed
+    loadLazyRouteModules(lazyMatches, manifest).then(() => {
+      let initialized =
+        !state.matches.some((m) => m.route.loader) ||
+        init.hydrationData != null;
+      if (initialized) {
+        // We already have required loaderData so we can just set initialized
+        updateState({ initialized: true });
+      } else {
+        // We still need to kick off initial data loads
+        startNavigation(HistoryAction.Pop, state.location);
+      }
+    });
```

Comment:
`initialize()` handles loading lazy module for any initial matches - this way we just keep rendering the `RouteProvider` `fallback` element